workers/data_refinery_workers/processors/test_salmon.py

```diff
@@ -0,0 +1,276 @@
+                         "Exception caught while retrieving raw file ERR003000_2.fastq.gz")
+
+        self.assertFalse(os.path.isfile(batch.files[0].get_temp_pre_path()))
+
+    def test_download_index_not_found(self):
+        batch, _, _ = init_objects()
+        processor_job = ProcessorJob.create_job_and_relationships(batches=[batch])
+
+        job_context = utils.start_job({"job": processor_job,
+                                       "job_id": processor_job.id,
+                                       "kmer_size": "23"})
+        job_context = salmon._prepare_files(job_context)
+
+        # Function we're testing.
+        salmon._download_index(job_context)
+
+        self.assertFalse(job_context["success"])
+        self.assertEqual(processor_job.failure_reason,
+                         "Failed to find an index for organism HOMO SAPIENS with kmer_size of 23.")
+
+    @patch.object(File, "download_processed_file")
+    def test_download_index_missing(self, mock_download_processed_file):
+        batch, _, _ = init_objects()
+        _insert_salmon_index()
+        processor_job = ProcessorJob.create_job_and_relationships(batches=[batch])
+
+        job_context = utils.start_job({"job": processor_job,
+                                       "job_id": processor_job.id,
+                                       "job_dir_prefix": "dummy",
+                                       "kmer_size": "23"})
+        job_context = salmon._prepare_files(job_context)
+
+        mock_download_processed_file.side_effect = FileNotFoundError()
+
+        # The function being testing.
+        salmon._download_index(job_context)
+
+        self.assertFalse(job_context["success"])
+        self.assertEqual(processor_job.failure_reason,
+                         "Failed to download and extract index tarball Homo_sapiens_short.gtf.gz")
+
+    def test_run_salmon_failure(self):
+        batch, first_fastq_file, second_fastq_file = init_objects()
+        processor_job = ProcessorJob.create_job_and_relationships(batches=[batch])
+
+        # Mock out the job_context with everything the function under
+        # test will expect
+        input_file_path_1 = first_fastq_file.get_temp_pre_path("dummy")
+        input_file_path_2 = second_fastq_file.get_temp_pre_path("dummy")
+        job_context = utils.start_job({"job": processor_job,
+                                       "job_id": processor_job.id,
+                                       "job_dir_prefix": "dummy",
+                                       "batches": [batch],
+                                       "index_directory": "missing",
+                                       "input_file_path": input_file_path_1,
+                                       "input_file_path_2": input_file_path_2,
+                                       "output_directory": "blah"})
+
+        # The function being tested.
+        job_context = salmon._run_salmon(job_context)
+
+        self.maxDiff = None
```

Comment:
Ah, what is this `self.maxDiff` being used for? I don't see it anywhere else.

Reply:
Good catch, I think I was using it at some point and then forgot to remove it once I no longer was.